Update early-access-nft-with-typescript.mdx (#56)
nftlabs/portal: guides/connect-to-blockchain/connect-to-blockchain.mdx

---
title: "Connect to the blockchain and use our SDK"
author: "Raza Zaidi"
authorAvatar: "/assets/portal/authors/raza-zaidi.jpg"
authorTwitter: "https://twitter.com/razacodes"
draft: false
tags: [sdk, client, server]
summary: "Providers and Signers are used to connect with the blockchain. Find out how to do this to use our sdk"
image: "./thumbnail.png"
date: "2022-02-03"
updated: ""
---

## Intro

> 

Reviewer: ```suggestion
```

Connecting and interacting with a blockchain from the frontend works slightly different than you might be used too. In this guide we will dive deeper into what type of connection you need to perform certain interactions with a blockchain.

## Provider

> ---

Reviewer: ```suggestion
---
```

delete all of the dividers under each title

# Using your own signer with thirdweb

---

**warning: this method is not suitable for frontend**

There are cases where you might want to run a script or integrate our sdk on the backend. In this case the `Connect Wallet` frontend component cannot be used.

Another way to connect to the blockchain in this case, is by passing the `Private Key` of your wallet directly, when instantiating the sdk. To understand more about **how** to do this properly (i.e. making sure your `Private Key` is not exposed), check out [this](/portal/guides/sdk-guide) guide.

When opting for this method to establish a connection to the blockchain, you will need to explicitly pass the RPC url too, to tell the sdk which chain you want to connect.

Here’s an example of doing this in TypeScript👇

```jsx
// Instantiate thirdrdweb SDK
const sdk = new ThirdwebSDK(
    new ethers.Wallet(
        // Your wallet private key
        process.env.PRIVATE_KEY as string,
        // RPC URL
        ethers.getDefaultProvider("https://rinkeby-light.eth.linkpool.io/")
    )
);
```

And here is one in Python 👇

```python
# choose your network
network = "https://rinkeby-light.eth.linkpool.io/"
# create the object

> sdk = ThirdwebSdk(SdkOptions(), network)

Reviewer: where is `SdkOptions()* coming from?

shouldn't we also import `ThirdwebSdk`?

---
title: "Connect to the blockchain and use our SDK"
author: "Raza Zaidi"
authorAvatar: "/assets/portal/authors/raza-zaidi.jpg"
authorTwitter: "https://twitter.com/razacodes"
draft: false
tags: [sdk, client, server]
summary: "Providers and Signers are used to connect with the blockchain. Find out how to do this to use our sdk"
image: "./thumbnail.png"
date: "2022-02-03"
updated: ""
---

## Intro

> ---

Reviewer: ```suggestion

```

why are we using dividers here? can we get rid of all of them?

## Provider

---

The provider is an easy way to connect to a block chain. When you _just_ want to **read** data from the blockchain, the provider is the best way to connect your app to the blockchain.

Regardless of using thirdweb’s sdk or another sdk, you will need a provider to establish this connection.

So a provider is a simple connection and this is what the connection looks like in terms of code:

```jsx
import { ethers } from ethers
const rpcUrl = "https://polygon-rpc.com/";
const provider = ethers.getDefaultProvider(rpcUrl);
```

A `provider` contains the information to tell our sdk which blockchain you want to use. A **RPC** url is the information the `provider` needs to decide which blockchain to retrieve data from.

**RPC** is a simple form of API interaction and stands for **Remote Procedure Call.** In other words we will use this url to make an API call to the chain.

# Using a read only provider with thirdweb

---

The thirdwebs sdk can be used to interact with your modules. To use our sdk to read data from the blockchain we need to create a connection with a `provider`. Then we tell it which blockchain to target via the **RPC** url.

> Make sure you have the packages installed. `npm install @3rdweb/sdk` for js and `pip install thirdweb-sdk` for python.

Reviewer: ```suggestion
Make sure you have the packages installed. `npm install @3rdweb/sdk` for JavaScript or TypeScript and `pip install thirdweb-sdk` for Python.
```

```jsx
import { ethers } from "ethers";
import { ThirdwebSDK } from "@3rdweb/sdk";

// This url indicates which chain you want to connect to
const rpcUrl = "https://polygon-rpc.com/";
const provider = ethers.getDefaultProvider(rpcUrl);
const sdk = new ThirdwebSDK(provider);
```

Here’s an example of how to use the provider in an use case where you _just_ want to retrieve data, like getting the number of NFTs you’ve minted so far. \*\*

```jsx
// Initialize nft module by passing in the module address

> nft_module_address = "0xD9bC403529471515F16DA32eee652c0a6cEcBA78";

Reviewer: ```suggestion
const nftModuleAddress = "0xD9bC403529471515F16DA32eee652c0a6cEcBA78";
```

Reviewer: camelCase

> const nft = sdk.getNFTModule(nft_module_address);

Reviewer: ```suggestion
const nft = sdk.getNFTModule(nftModuleAddress);
```

Another way to connect to the blockchain in this case, is by passing the `Private Key` of your wallet directly, when instantiating the sdk. To understand more about **how** to do this properly (i.e. making sure your `Private Key` is not exposed), check out [this](/portal/guides/sdk-guide) guide.

When opting for this method to establish a connection to the blockchain, you will need to explicitly pass the RPC url too, to tell the sdk which chain you want to connect.

Here’s an example of doing this in TypeScript👇

```jsx
// Instantiate thirdrdweb SDK
const sdk = new ThirdwebSDK(
    new ethers.Wallet(
        // Your wallet private key
        process.env.PRIVATE_KEY as string,
        // RPC URL
        ethers.getDefaultProvider("https://rinkeby-light.eth.linkpool.io/")
    )
);
```

And here is one in Python 👇

```python
# choose your network
network = "https://rinkeby-light.eth.linkpool.io/"
# create the object
sdk = ThirdwebSdk(SdkOptions(), network)
# use the '.env' file inside Python
loadenv("./.env")
PRIVATE_KEY = os.getenv('PRIVATE_KEY')
# over here we connect our wallet via the private key stored in the '.env' file

> sdk.set_private_key("PRIVATE_KEY")

Reviewer: ```suggestion
sdk.set_private_key(PRIVATE_KEY)
```

variable call not string
